tortoise/backends/base/schema_generator.py

```diff
@@ -324,22 +338,30 @@ def _get_table_sql(self, model: "Type[Model]", safe: bool = True) -> dict:
                     self._get_unique_constraint_sql(model, unique_together_to_create)
                 )
 
-        # Indexes.
         _indexes = [
             self._get_index_sql(model, [field_name], safe=safe) for field_name in fields_with_index
         ]
 
         if model._meta.indexes:
-            for indexes_list in model._meta.indexes:
-                if not isinstance(indexes_list, Index):
-                    indexes_to_create = []
-                    for field in indexes_list:
+            for index in model._meta.indexes:
+                if not isinstance(index, Index):
+                    fields = []
+                    for field in index:
                         field_object = model._meta.fields_map[field]
-                        indexes_to_create.append(field_object.source_field or field)
+                        fields.append(field_object.source_field or field)
 
-                    _indexes.append(self._get_index_sql(model, indexes_to_create, safe=safe))
+                    _indexes.append(self._get_index_sql(model, fields, safe=safe))
                 else:
-                    _indexes.append(indexes_list.get_sql(self, model, safe))
+                    if index.fields:
+                        fields = [f for f in index.fields]
+                    else:
+                        fields = [f"({expression.get_sql()})" for expression in index.expressions]
```

Comment:
Maybe it would be a little bit clearer with
```
                    if index.fields:
                        fields = [f for f in index.fields]
                    elif index.expressions:
                        fields = [f"({expression.get_sql()})" for expression in index.expressions]
                     else:
                           raise RuntimeError("Invalid index object")
```

Not that much of difference, but could save some headache in case of future refactorings or something like that

Reply:
I added an extra check.